### PR TITLE
System Overview docs

### PR DESCRIPTION
**Pull request by dannyroberts**

### CODE REVIEW

File: docs/system-overview.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# System Overview
+
+Commcare Cloud gives you the tools you need to go from an inventory of virtual machines
+to a robust production setup adhering to Dimagi's own security standards and best practices.
```
Comment (esoergel) [inline docs/system-overview.md]: What a hook!

File: docs/system-overview.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+and that there be no other method of SSH login enabled,
+and these policies are automatically implemented via the `commcare-cloud` `update-users`
+command (or the `deploy-stack` command, which includes the former).
+
+As a result of these policies, it's important to keep in mind that
+- Password-based log-ins are disabled for all users
+- Every individual who needs to be able to log in must have an SSH key pair,
+  and their public key must be recorded in your environment's configuration files.
+- There are no shared log-ins, and thus no root login.
+  All access to shared user accounts will be mediated by the individual's own SSH key log-in.
```
Comment (esoergel) [inline docs/system-overview.md]: Missing a `-` here

File: docs/system-overview.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Dimagi has a strict policy that every individual log in using their own SSH key pair,
+and that there be no other method of SSH login enabled,
+and these policies are automatically implemented via the `commcare-cloud` `update-users`
+command (or the `deploy-stack` command, which includes the former).
+
+As a result of these policies, it's important to keep in mind that
+- Password-based log-ins are disabled for all users
+- Every individual who needs to be able to log in must have an SSH key pair,
+  and their public key must be recorded in your environment's configuration files.
+- There are no shared log-ins, and thus no root login.
```
Comment (esoergel) [inline docs/system-overview.md]: What about `ansible`?  Isn't that arguably a shared login?  It's described as such below.

File: docs/system-overview.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+- Individuals' users do not have passwords associated with them.
+
+In addition to logging in as their own user, each individual can log in
+as the special `ansible` user, because their public key will be present in that
+user's `authorized_keys` file as well. This user has `sudo` privileges
+as well as a password that can be used to `sudo` or become any other user via `su`.
+
+Each individual user also has the ability to run a small number of `sudo` commands
+without a password; notably, `sudo -u cchq bash` and `sudo -u ansible bash`,
+letting them become the `cchq` or `ansible` user for convenience.
```
Comment (esoergel) [inline docs/system-overview.md]: Might be worth a note about why it's necessary that users be able to `sudo -u` to `cchq` or `ansible`

File: docs/system-overview.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+### Encrypted Drive
+
+Every machine that runs a stateful service (such as a database)
+will have an encrypted drive using [ecryptfs](http://ecryptfs.org/).
+CommCare Cloud will configure this drive for you,
+but it is important that you are aware of it, notably because it must be mounted
+with its password each time the VM is rebooted before it can be used.
+The mounting process is automated as well and will be run as part of the
+`commcare-cloud` `after-reboot` command.
```
Comment (esoergel) [inline docs/system-overview.md]: Nit: You can auto wrap text to a specified column width using `gq` in vim (`set textwidth=80` to change width), or "Fill Paragraph" in PyCharm.   Just an FYI if you're doing it manually.

File: docs/system-overview.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+---
+
+[︎⬅︎ Overview](.)
```
Comment (esoergel) [inline docs/system-overview.md]: neat!

File: docs/system-overview.md (hunk 2 above)
Comment (esoergel) [reply docs/system-overview.md]: Oh, nevermind - that's the same bullet

File: docs/system-overview.md (hunk 3 above)
Comment (dannyroberts) [reply docs/system-overview.md]: I was using log-in here to mean authentication. I.e. at the ssh step no shared credential is enough to give someone ssh login access. I'll change this to say no shared credentials and no root login; you're right that this wasn't very clear.

**Pull request merged**